### ISSUE TITLE
Breaking change to passing multiple properties as /p:P1=V1;P2=V2

### ISSUE THREAD

**Issue by nguerrera** [dotnet/cli#9369]
I believe this was changed by https://github.com/dotnet/cli/pull/9153

## Steps to reproduce

1. Create test.proj with the following content
``` xml
<Project>
  <Target Name="Build">
    <Message Importance="High" Text="A: $(A)" />
    <Message Importance="High" Text="B: $(B)" />
  </Target>
</Project>
```

2. `dotnet build test.proj /p:A=1;B=2 /nologo /clp:NoSummary`

## Expected  behavior
```
  A: 1
  B: 2
```

## Actual behavior
```
  A: 1;B=2
  B:
```

## Environment data
```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.100-preview1-008941
 Commit:    dae5099ffb

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   d:\src\s\sdk\.dotnet\sdk\2.2.100-preview1-008941\

Host (useful for support):
  Version: 2.1.0-rtm-26515-03
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.2.100-preview1-008941 [d:\src\s\sdk\.dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0-rtm-30752 [d:\src\s\sdk\.dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0-rtm-30752 [d:\src\s\sdk\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.5 [d:\src\s\sdk\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [d:\src\s\sdk\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.0 [d:\src\s\sdk\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0-rtm-26515-03 [d:\src\s\sdk\.dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by nguerrera** [comment]
cc @peterhuene

**Comment by peterhuene** [comment]
Sigh.  I didn't know MSBuild supported that.  We may need a fix to the command line parser that enables us to know if the original command line argument was quoted or not.  Otherwise, I don't know how we can fix this to support both uses (unquoted, a delimiter for multiple properties; quoted, a literal semicolon).

**Comment by nguerrera** [comment]
cc @rainersigwald

**Comment by rainersigwald** [comment]
I don't think it can be fixed. On Windows/full framework, MSBuild inspects the raw command line string and handles quoting itself to get the intuitive behavior (`/p:Foo="bar;baz" /p:Bat=z;Pow=q` results in 3 properties). But that's not possible in unix-likes where the quoting is handled by the shell and the program only has access to an argument array.

**Comment by rainersigwald** [comment]
Discussion at https://github.com/Microsoft/msbuild/issues/471.

**Comment by peterhuene** [comment]
Ah, I see the crux of the problem then.  If MSBuild decided on [option 4](https://github.com/Microsoft/msbuild/issues/471#issuecomment-183053180), then I'd be in favor of reverting #9153 and ask users to do the same for `dotnet` (escape quotes around the property value) and to not try anything fancy from our side.

**Comment by rainersigwald** [comment]
I'm ready to commit to that. I think it's the least-bad option. @AndyGerlicher objections?

**Comment by AndyGerlicher** [comment]
Seems reasonable given no good options.

**Comment by peterhuene** [comment]
Ok, I'll revert and then comment on the CLI issue to let users know.  Thanks everyone!

**Comment by peterhuene** [comment]
Fixed with #9394.
